appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/UberJarCreator.java

```diff
@@ -241,6 +251,19 @@ public void buildUberJar() {
                     }
                 }
             }
+            
+            if (copyDirectory != null) {
+                String basePath = copyDirectory.getPath().replaceAll(copyDirectory + "$", "");
+                List<File> filesToCopy = fillFiles(copyDirectory);
+                for (File file : filesToCopy) {
+                    
+                        JarEntry deploymentEntry = new JarEntry(copyDirectory + file.getPath().replace(basePath, ""));
+                        jos.putNextEntry(deploymentEntry);
+                        Files.copy(file, jos);
```

Comment:
the 3rd party jar does not need to be uncompressed but the zip entry storing it must be if placed in MICRO-INF/runtime or MICRO-INF/lib. However this is only a requirement if running with the nested jar classloader which be default Payara Micro doesn't